src/TinyMessenger/TinyMessenger.cs

```diff
@@ -802,10 +802,10 @@ where sub.Subscription.ShouldAttemptDelivery(message)
         #else
         private async void PublishAsyncInternal<TMessage>(TMessage message, AsyncCallback callback) where TMessage : class, ITinyMessage
         {
-            Action publishAction = () => { PublishInternal<TMessage>(message); };
-
-            await Task.Run(publishAction.Invoke);
-            PublishInternal<TMessage>(message);

```

Comment:
Can we simplify this into something like this? We should keep this simple and Publish Internal doesn't need to be async as it's a synchronous operation and can run inline (the callback can be awaited).

```cs
PublishInternal<TMessage>(message);
// invoke callback here?
```

Reply:
You right! Since the `async void` can't be awaited and is asynchronously therefor. I personally don't like `async void` at all but in this case it might make sense to use it for readability and performance reasons.

Reply:
Prior to the mistake in my PR, the original code was async via BeginInvoke (and still is for the not NETSTANDARD case).  As I understand it, BeginInvoke queues to a Thread Pool. The only way to be synchronous is to “join” with the EndInvoke method.

Unless you want to change that aspect of the Publish method, I think something needs to occur to avoid callbacks blocking the caller thread.

https://devblogs.microsoft.com/dotnet/migrating-delegate-begininvoke-calls-for-net-core/

Reply:
@niemyjski, @gkarabin. Thanks for the quick and good feedback. I've tried to simplify the method as much as possible while keeping the interface. With the current `void` method I don't think there's a simpler way than using `Task.Run` for doing the publishing asynchronously.

Having `async Tasks` with cancellation support for asynchronous methods would be great but the refactoring would exceed the scope of a #148 bugfix for sure.

Reply:
Thanks for the link! I think this looks good with the current signature (fire and forget). Are you happy with me merging this pr?

Reply:
It looks good to me!